app/src/main/java/com/fmpdroid/moneybox/CreateMoneyBox.java

```diff
@@ -0,0 +1,94 @@
+
+import com.fmpdroid.moneybox.dto.MoneyBoxDto;
+import com.fmpdroid.moneybox.dto.Singleton;
+import com.google.android.material.textfield.TextInputEditText;
+
+import java.util.Calendar;
+import java.util.Locale;
+
+public class CreateMoneyBox extends AppCompatActivity {
+
+    private Context mContext = CreateMoneyBox.this;
+    private Calendar mCalendar = Calendar.getInstance();
+    private TextInputEditText edtTime;
+    private TextInputEditText edtDate;
+    private TextInputEditText edtTitle;
+    private TextInputEditText edtDescription;
+    private TextInputEditText edtTargetAmount;
+    private Switch mSwitch;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.activity_create_money_box);
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setTitle(R.string.title_activity_create_money_box);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        getSupportActionBar().setDisplayShowHomeEnabled(true);
+
+        edtTime = findViewById(R.id.edtTime);
+        edtDate = findViewById(R.id.edtDate);
+        edtTitle = findViewById(R.id.edtMoneyBoxName);
+        edtDescription = findViewById(R.id.edtDescription);
+        edtTargetAmount = findViewById(R.id.edtTargetAmount);
+        mSwitch = findViewById(R.id.switchAllowReminder);
+
+
+        edtTime.setText(String.format("%02d:%02d", mCalendar.get(Calendar.HOUR_OF_DAY), mCalendar.get(Calendar.MINUTE)));
+        edtDate.setText(mCalendar.getDisplayName(Calendar.MONTH, Calendar.LONG, Locale.getDefault()) + " " + mCalendar.get(Calendar.DAY_OF_MONTH) + ", " + mCalendar.get(Calendar.YEAR));
+    }
+
+    public void createMoneyBox(View v) {
+
+        Intent intent = new Intent(CreateMoneyBox.this, MainActivity.class);
+        intent.putExtra(this.getResources().getString(R.string.key_isEmpty), false);
+        MoneyBoxDto moneyBox = new MoneyBoxDto();
+        moneyBox.setAllowReminder(mSwitch.isChecked());
+        moneyBox.setDateCreated(edtDate.getText().toString());
+        moneyBox.setDescription(edtDescription.getText().toString());
+        moneyBox.setTargetAmount(Float.parseFloat(edtTargetAmount.getText().toString()));
+        moneyBox.setTargetDate(edtDate.getText().toString());
+        moneyBox.setTitle(edtTitle.getText().toString());
+        Singleton singleton = Singleton.getInstance();
+        singleton.addMoneyBox(moneyBox);
+        startActivity(intent);
+    }
+
+    public void pickTime(View v) {
+
+        TimePickerDialog timePickerDialog = new TimePickerDialog(mContext, (timePicker, hourOfDay, minute) ->
+                edtTime.setText(String.format("%02d:%02d", hourOfDay, minute)), mCalendar.get(Calendar.HOUR_OF_DAY),
+                mCalendar.get(Calendar.MINUTE), android.text.format.DateFormat.is24HourFormat(mContext));
+        timePickerDialog.show();
```

Comment:
No need to put this in a variable. directly call the show() method.

```suggestion
        new TimePickerDialog(mContext, (timePicker, hourOfDay, minute) ->
                edtTime.setText(String.format("%02d:%02d", hourOfDay, minute)), mCalendar.get(Calendar.HOUR_OF_DAY),
                mCalendar.get(Calendar.MINUTE), android.text.format.DateFormat.is24HourFormat(mContext)).show();
```